components/Map.js

```diff
@@ -52,21 +60,149 @@ class Map extends React.Component {
+      });
+
+      this.map.addLayer({
+        id: 'publicWorksRamps',
+        type: 'circle',
+        source: 'publicWorksRamps-point',
+        paint: {
+          'circle-stroke-width': 1,
+          'circle-stroke-color': '#091F2F',
+          'circle-color': '#c1c1c1',
+          // Make circles larger as the user zooms from 12 to 22.
+          'circle-radius': {
+            base: 3,
+            stops: [[12, 3], [17, 10]],
+          },
+        },
+      });
+
+      // Add the budget facilities as a layer.
+      this.map.addSource('budgetFacilities-point', {
         type: 'geojson',
         data: `${projects_url}/query?where=1%3D1&outFields=*&outSR=4326&returnExceededLimitFeatures=true&f=pgeojson`,
       });
 
       this.map.addLayer({
-        id: 'projects-point',
+        id: 'budgetFacilities',
         type: 'circle',
-        source: 'projects',
+        source: 'budgetFacilities-point',
         paint: {
           'circle-stroke-width': 1,
+          'circle-stroke-color': '#091F2F',
+          'circle-color': '#288BE4',
+          // Make circles larger as the user zooms from 12 to 22.
+          'circle-radius': {
+            base: 4,
+            stops: [[12, 4], [17, 10]],
+          },
         },
       });
     });
+
+    // Add pop-ups to map
+    this.map.on('click', e => {
+      // Budget aggregates the data so there is only one project per
+      // feature (no projects overlap eachother), so we only grab the
+      // first feature clicked on.
+      const feature = this.map.queryRenderedFeatures(e.point)[0];
+      const dataset = feature.layer.id;
+      // Feature properties can differ based on the data selected.
+      const properties =
+        dataset == 'budgetFacilities'
+          ? [
+              feature.properties.Project_Title,
+              feature.properties.Project_Description,
+              feature.properties.Publish_Status,
+              feature.properties.Total_Budget,
+              feature.properties.Department,
+              feature.properties.Cabinet,
+            ]
+          : [
+              // Set the properties for publicWorksRamps.
+              feature.properties.Street,
+              feature.properties.NAME,
+              feature.properties.CITYCODE,
+              feature.properties.CONDITION,
+            ];
+
+      new mapboxgl.Popup({ closeOnClick: true })
+        .setLngLat(feature.geometry.coordinates)
+        // We create HTML elements for the populated properties.
+        .setHTML(
+          `<div style="min-width: 200px; max-width: 250px">
+        <div>
+            <ul class="dl dl--sm">
+              <li class="dl-i dl-i--b">
+                <div class="dl-d">${properties[0]}</div>
+              </li>        
+              <li class="dl-i dl-i--b">
+                <div class="dl-t">
+                ${properties[2] ? `${properties[2]}<br/>` : ''}
```

Comment:
This is slightly janky to manually enumerate up to 5. You could take the array of properties, filter out any that are empty, and `join` with`"<br/>"`.

Reply:
Good point. Made this better also realized I didn't even need to filter them - can just map through them. Improving styling for these will be in next PR.